supply/lib/supply/setup.rb

```diff
@@ -48,32 +48,32 @@ def download_images(listing)
 
       allowed_imagetypes.each do |image_type|
         begin
+          image_counter = 1 # Used to prefix the downloaded files, so order is preserved.
+          if IMAGES_TYPES.include?(image_type) # IMAGE_TYPES are stored in locale/images location
+            path = File.join(metadata_path, listing.language, IMAGES_FOLDER_NAME, image_type.to_s)
+          else # All other screenshot types go under their respective folders.
+            path = File.join(metadata_path, listing.language, IMAGES_FOLDER_NAME, image_type, "#{image_counter}_#{listing.language}")
```

Comment:
this won't work outside of the `urls.each` loop though, right?

Reply:
I moved it to the beginning of `allowed_imagetypes` loop, so it creates folders for all image types. If `urls` is `null`, nothing is downloaded (but the folder is still created). 

Hope this is what you are looking for. Let me know if you are looking for something else, please :)

Reply:
`image_counter` in `path` will always be `1` now, as it is only incremented in line 82 (inside the loop).

I think you have to separate the folder and filename parts of the path - the folder can be defined outside, but the filename has to be inside of the loop.

Reply:
Ah, I see what you mean :)

If there are no images to download, do we need to increment the counter? (And if there are images, it would be incremented.).`FileUtils.mkdir_p(p.dirname.to_s)` will create the parent directory.

Reply:
> (And if there are images, it would be incremented.)

Sure?

The path is defined only once now, but we are saving many images. The variable will be incremented, but the path will not be updated. (Or did I miss something?)

Reply:
ah I see, my bad 🤦‍♂️ fixing it now :)